### PR TITLE
Correct `Nullable<T>.Equals()` calls in `BoundNode.Update()`

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs [hunk 1]
```diff
@@ -1013,7 +1013,7 @@ string wasUpdatedCheck(Field field)
                                 : TypeIsSymbol(field)
                                     ? "!Symbols.SymbolEqualityComparer.ConsiderEverything.Equals({0}, this.{1})"
                                     : IsValueType(field.Type) && field.Type[^1] == '?'
-                                        ? "{0}.Equals(this.{1})"
+                                        ? "!{0}.Equals(this.{1})"
```
Comment (cston) [inline src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs:1016]: `Nullable<T>.Equals(object)` requires boxing the argument. Perhaps we should use an alternative.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs [hunk 2]
```diff
@@ -1956,6 +1960,26 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                 return null;
             }
         }
+
+#if DEBUG
+        private static bool IsEmptyRewritePossible(BoundNode node)
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs:1965]: Consider just marking this method/class with `Conditional` and removing the `#if`s

**Comment by 333fred** [comment]
Was this just something you noticed, or was there a test failure or bug somewhere?

**Comment by cston** [comment]
> Was this just something you noticed, or was there a test failure or bug somewhere?

I noticed this when adding another `Nullable<T>` field to a `BoundNode` type which did hit a test failure.

**Review by 333fred — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs (hunk 2 above)
Comment (jcouv) [reply src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs:1965]: nit: FWIW, switching to a conditional `void` method "AssertEmptyRewritePossible" would have solved that

File: src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs [hunk 3]
```diff
@@ -1828,6 +1828,10 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     }
                     forSemanticModel = new MethodBodySemanticModel.InitialState(syntaxNode, methodBodyForSemanticModel, bodyBinder, snapshotManager, remappedSymbols);
 
+#if DEBUG
+                    Debug.Assert(IsEmptyRewritePossible(methodBody));
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs:1832]: >Debug.Assert(IsEmptyRewritePossible(methodBody)); [](http://example.com/codeflow?start=20&length=49)

Is it worth adding similar check for VB?

**Pull request merged**